src/commands/uploadFiles/UploadFilesStep.ts

```diff
@@ -0,0 +1,143 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardExecuteStep, IActionContext, IParsedError, nonNullValue, parseError } from "@microsoft/vscode-azext-utils";
+import { CancellationToken, ProgressLocation, Uri, window } from "vscode";
+import { NotificationProgress, storageFilter } from '../../constants';
+import { ext } from "../../extensionVariables";
+import { BlobContainerTreeItem } from "../../tree/blob/BlobContainerTreeItem";
+import { FileShareTreeItem } from "../../tree/fileShare/FileShareTreeItem";
+import { AzExtFsExtra } from "../../utils/AzExtFsExtra";
+import { isAzCopyError, multipleAzCopyErrorsMessage, throwIfCanceled } from "../../utils/errorUtils";
+import { localize } from "../../utils/localize";
+import { checkCanUpload, convertLocalPathToRemotePath, getUploadingMessage, outputAndCopyUploadedFileUrls, OverwriteChoice, upload } from "../../utils/uploadUtils";
+import { IAzCopyResolution } from "../azCopy/IAzCopyResolution";
+import { IExistingFileContext } from "./IExistingFileContext";
+import { IUploadFilesWizardContext } from './IUploadFilesWizardContext';
+
+let lastUriUpload: Uri | undefined;
+
+export class UploadFilesStep extends AzureWizardExecuteStep<IUploadFilesWizardContext> {
+    public priority: number = 100;
+    public treeItem?: BlobContainerTreeItem | FileShareTreeItem;
+    public uris?: Uri[];
+    public notificationProgress?: NotificationProgress;
+    public cancellationToken?: CancellationToken;
+
+    public constructor(treeItem?: BlobContainerTreeItem | FileShareTreeItem, uris?: Uri[], notificationProgress?: NotificationProgress, cancellationToken?: CancellationToken) {
+        super();
+        this.treeItem = treeItem;
+        this.uris = uris;
+        this.notificationProgress = notificationProgress;
+        this.cancellationToken = cancellationToken;
+    }
+
+    public async execute(context: IUploadFilesWizardContext, _progress: NotificationProgress): Promise<void> {
+        const calledFromUploadToAzureStorage: boolean = this.uris !== undefined;
```

Comment:
Is `[]` (empty array) semantically equivalent to `undefined` in this context? If so, I'd suggest:
```suggestion
        const calledFromUploadToAzureStorage: boolean = !!this.uris?.length;
```